### ISSUE TITLE
undefined requests from webapp towards server

### ISSUE THREAD

**Issue by zetaab** [mattermost/mattermost-server#9445]
#### Summary
webapp is requesting following urls quite often:
```
/api/v4/files/undefined/thumbnail
/api/v4/files/undefined/preview
/api/v4/channels/undefined/posts
````

Example
```
{"level":"error","ts":1537768824.6891952,"caller":"web/context.go:60","msg":"Invalid or missing file_id parameter in request URL","path":"/api/v4/files/undefined/thumbnail","request_id":"xrxu3zyh4jbpmgy81p7je66n6y","ip_addr":"xxx","user_id":"kzkpdqnkejfpfpye1i5qtdctco","method":"GET","err_where":"Context","http_code":400,"err_details":""}
{"level":"error","ts":1537769538.9725723,"caller":"web/context.go:60","msg":"Invalid or missing file_id parameter in request URL","path":"/api/v4/files/undefined/thumbnail","request_id":"ejhj5hp3xfdzd8gq4i6g1ed69w","ip_addr":"xxx","user_id":"qzocuc9a3ibr7fadib9kroxsac","method":"GET","err_where":"Context","http_code":400,"err_details":""}
{"level":"error","ts":1537769834.3792932,"caller":"web/context.go:60","msg":"Invalid or missing channel_id parameter in request URL","path":"/api/v4/channels/undefined/posts","request_id":"e5wj89k5fbgjpxk83radbdc78a","ip_addr":"xxx","user_id":"n93b5gim53dytkmh19jaoggxbe","method":"GET","err_where":"Context","http_code":400,"err_details":""}
{"level":"error","ts":1537769866.0723476,"caller":"web/context.go:60","msg":"Invalid or missing file_id parameter in request URL","path":"/api/v4/files/undefined/thumbnail","request_id":"3p3khfnh7jnedpdiom9u5xhjzw","ip_addr":"xx","user_id":"8we9uid4hfr3iywagra5qinymr","method":"GET","err_where":"Context","http_code":400,"err_details":""}
{"level":"error","ts":1537769866.5333483,"caller":"web/context.go:60","msg":"Invalid or missing file_id parameter in request URL","path":"/api/v4/files/undefined/preview","request_id":"pmmqtj96wj8djnyw9r1t65f3ro","ip_addr":"xxx","user_id":"8we9uid4hfr3iywagra5qinymr","method":"GET","err_where":"Context","http_code":400,"err_details":""}
{"level":"error","ts":1537769933.9002864,"caller":"web/context.go:60","msg":"Invalid or missing channel_id parameter in request URL","path":"/api/v4/channels/undefined/posts","request_id":"rwa6rbx5i7b4dqw7wuyro4n9kr","ip_addr":"xxx","user_id":"mxuc6i4qab8kfpcdmdtjtkyi6h","method":"GET","err_where":"Context","http_code":400,"err_details":""}
{"level":"error","ts":1537770194.1643252,"caller":"web/context.go:60","msg":"Invalid or missing channel_id parameter in request URL","path":"/api/v4/channels/undefined/posts","request_id":"ntqgfxp9ki8gicpc8nne49uyyc","ip_addr":"xxx","user_id":"8i1isfgqutgype115a4s94n3oo","method":"GET","err_where":"Context","http_code":400,"err_details":""}
{"level":"error","ts":1537770700.7889876,"caller":"web/context.go:60","msg":"Invalid or missing channel_id parameter in request URL","path":"/api/v4/channels/undefined/posts","request_id":"t6m8n3shutdibedww5g6kbdjsy","ip_addr":"xxx","user_id":"knwzx6pin78q5efgjtm9asfxfw","method":"GET","err_where":"Context","http_code":400,"err_details":""}
```

like you can think, these urls does not work. From last 4 days:
```
% oc logs mattermost-698b867c89-p6fnt|grep undefined|wc -l
     272
```
#### Steps to reproduce
mattermost 5.3.1, enough big mattermost installation (over 2000 users) and you will get much of these errors to log

#### Expected behavior
I except that webapp is not requesting urls at all, if id is undefined

**Comment by amyblais** [comment]
Hi @zetaab, thank you for reporting this.

I observed the logs on a few of our servers and didn't see the same pattern - are you able to think of a specific situation / reproduction steps when you start seeing these log errors?

Also, do you have this setting disabled: System Console > Posts > Enable Link Preview? There is a bug related to this setting and I'm wondering if the issue you are seeing is related to that: https://mattermost.atlassian.net/browse/MM-12264.

**Comment by amyblais** [comment]
This issue has been resolved and tested internally (https://mattermost.atlassian.net/browse/MM-12264). The fix will be available in v5.4 release. Please re-open this ticket if I'm wrong / misunderstood the issue here,
